pkg/config/config.go

```diff
@@ -249,3 +252,25 @@ func KeyAsEnvVar(key string) string {
 		fmt.Sprintf("%s_%s", environmentVariablePrefix, environmentVariableReplace.Replace(key)),
 	)
 }
+
+func GetNodeID(ctx context.Context, nodeNameProviderType string) (string, error) {
```

Comment:
nit: better to name it GenerateNodeID to avoid confusing it with getting the actual node name